modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIFilterRelevantObjectEntryModelListener.java

```diff
@@ -51,15 +54,33 @@ private void _validate(ObjectEntry objectEntry) {
 		try {
 			Map<String, Serializable> values = objectEntry.getValues();
 
+			long apiEndpointId = (long)values.get(
+				"r_apiEndpointToAPIFilters_c_apiEndpointId");
+
 			if (!_objectEntryHelper.isValidObjectEntry(
-					(long)values.get(
-						"r_apiEndpointToAPIFilters_c_apiEndpointId"),
-					"L_API_ENDPOINT")) {
+					apiEndpointId, "L_API_ENDPOINT")) {
 
 				throw new ObjectEntryValuesException.InvalidObjectField(
 					null, "An API filter must be related to an API endpoint",
 					"an-api-filter-must-be-related-to-an-api-endpoint");
 			}
+
+			if (Validator.isNotNull(
+					_objectEntryHelper.getObjectEntry(
+						objectEntry.getCompanyId(),
+						StringBundler.concat(
+							"id ne '", objectEntry.getObjectEntryId(),
+							"' and r_apiEndpointToAPIFilters_c_apiEndpointId ",
+							"eq '", apiEndpointId, "'"),
+						getObjectDefinitionExternalReferenceCode()))) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					Collections.singletonList(apiEndpointId),
```

Comment:
I think that `Collections.singleList(apiEndpointId)` is not needed because you are using a message key where there isn't any argument to pass.

Reply:
Yes you're right. I wrote that because originally I was going to return the `apiEndpointId` in the response and I forgot to change it. Thanks!